jgscm/__init__.py

```diff
@@ -428,29 +441,50 @@ def client(self):
         """
         :return: used instance of :class:`google.cloud.storage.Client`.
         """
-        try:
-            return self._client
-        except AttributeError:
-            if not self.project:
-                self._client = GSClient()
-            else:
-                self._client = GSClient.from_service_account_json(
-                    self.keyfile, project=self.project)
+        if self._client is not None:
             return self._client
+        if not self.project:
+            self._client = GSClient()
+        else:
+            self._client = GSClient.from_service_account_json(
+                self.keyfile, project=self.project)
+        return self._client
 
     def run_post_save_hook(self, model, os_path):
         """Run the post-save hook if defined, and log errors"""
         if self.post_save_hook:
             try:
                 self.log.debug("Running post-save hook on %s", os_path)
-                self.post_save_hook(os_path=path, model=model, contents_manager=self)
+                self.post_save_hook(os_path=os_path,
+                                    model=model,
+                                    contents_manager=self)
             except Exception:
                 self.log.error("Post-save hook failed on %s", os_path, exc_info=True)
 
     @default("checkpoints_class")
     def _checkpoints_class_default(self):
         return GoogleStorageCheckpoints
 
+    def _resolve_storagetype(self, path, type):
```

Comment:
Could you please rename `type` to smth else - it clashes with the built-in name. I know that it does not matter much in this particular case, but still.

Reply:
Was named type to comply with existing arg-name convention. I agree that using a builtin is... unpleasant. Anyways. Fixed.